File: frontend/lib/models/today_status.dart
class CardEntryStatus {
  final int id;
  final double amount;
  final String? bankName;
  final String? lastFour;

  CardEntryStatus({
    required this.id,
    required this.amount,
    this.bankName,
    this.lastFour,
  });

  factory CardEntryStatus.fromJson(Map<String, dynamic> json) {
    return CardEntryStatus(
      id: json['id'] as int,
      amount: (json['amount'] as num).toDouble(),
      bankName: json['bank_name'] as String?,
      lastFour: json['last_four'] as String?,
    );
  }
}

class CashEntryStatus {
  final bool submitted;
  final double? amount;
  final int? entryId;

  CashEntryStatus({
    required this.submitted,
    this.amount,
    this.entryId,
  });

  factory CashEntryStatus.fromJson(Map<String, dynamic> json) {
    return CashEntryStatus(
      submitted: json['submitted'] as bool,
      amount:
          json['amount'] != null ? (json['amount'] as num).toDouble() : null,
      entryId: json['entry_id'] as int?,
    );
  }
}

class CardStatusToday {
  final int count;
  final double total;
  final List<CardEntryStatus> entries;

  CardStatusToday({
    required this.count,
    required this.total,
    required this.entries,
  });

  factory CardStatusToday.fromJson(Map<String, dynamic> json) {
    return CardStatusToday(
      count: json['count'] as int,
      total: (json['total'] as num).toDouble(),
      entries: (json['entries'] as List? ?? [])
          .map((e) => CardEntryStatus.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}

class TodayStatus {
  final String date;
  final CashEntryStatus mainCash;
  final CashEntryStatus manoCash;
  final CardStatusToday card;

  TodayStatus({
    required this.date,
    required this.mainCash,
    required this.manoCash,
    required this.card,
  });

  factory TodayStatus.fromJson(Map<String, dynamic> json) {
    return TodayStatus(
      date: json['date'] as String,
      mainCash:
          CashEntryStatus.fromJson(json['main_cash'] as Map<String, dynamic>),
      manoCash:
          CashEntryStatus.fromJson(json['mano_cash'] as Map<String, dynamic>),
      card: CardStatusToday.fromJson(json['card'] as Map<String, dynamic>),
    );
  }
}

File: ios/Runner/Models/TodayStatus.swift
import Foundation

struct CardEntryStatus: Codable, Identifiable {
    let id: Int
    let amount: Double
    let bankName: String?
    let lastFour: String?

    private enum CodingKeys: String, CodingKey {
        case id, amount
        case bankName = "bank_name"
        case lastFour = "last_four"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id       = try c.decode(Int.self, forKey: .id)
        bankName = try? c.decode(String.self, forKey: .bankName)
        lastFour = try? c.decode(String.self, forKey: .lastFour)
        let raw  = try c.decode(AnyCodable.self, forKey: .amount)
        amount   = Double("\(raw.value)") ?? 0
    }
}

struct CashEntryStatus: Codable {
    let submitted: Bool
    let amount: Double?
    let entryId: Int?

    private enum CodingKeys: String, CodingKey {
        case submitted, amount
        case entryId = "entry_id"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        submitted = try c.decode(Bool.self, forKey: .submitted)
        entryId   = try? c.decode(Int.self, forKey: .entryId)
        if let raw = try? c.decode(AnyCodable.self, forKey: .amount) {
            amount = Double("\(raw.value)")
        } else { amount = nil }
    }
}

struct CardStatusToday: Codable {
    let count: Int
    let total: Double
    let entries: [CardEntryStatus]

    private enum CodingKeys: String, CodingKey {
        case count, total, entries
    }

    init(from decoder: Decoder) throws {
        let c   = try decoder.container(keyedBy: CodingKeys.self)
        count   = try c.decode(Int.self, forKey: .count)
        entries = (try? c.decode([CardEntryStatus].self, forKey: .entries)) ?? []
        let raw = try c.decode(AnyCodable.self, forKey: .total)
        total   = Double("\(raw.value)") ?? 0
    }
}

struct TodayStatus: Codable {
    let date: String
    let mainCash: CashEntryStatus
    let manoCash: CashEntryStatus
    let card: CardStatusToday

    private enum CodingKeys: String, CodingKey {
        case date
        case mainCash = "main_cash"
        case manoCash = "mano_cash"
        case card
    }
}
